### ISSUE TITLE
Strange swap of XML leaf nodes

### ISSUE THREAD

**Issue by MoronVV** [Shoobx/xmldiff#36]
Comparing this 2 xmls:
```xml
<Root>
  <A>2019-10-30</A>
  <B>
    <C>
      <D>1948-07-31</D>
    </C>
  </B>
</Root>
```
```xml
<Root>
  <A>2020-01-15</A>
  <B>
    <C>
      <D>1948-07-31</D>
    </C>
  </B>
</Root>
```
I get:
```xml
<Root xmlns:diff="http://namespaces.shoobx.com/diff">
  <A diff:insert="">2020-01-15</A>
  <A diff:delete="">2019-10-30</A>
  <B>
    <C>
      <D diff:insert="" diff:rename="A">1948-07-31</D>
      <D diff:delete="">1948-07-31</D>
    </C>
  </B>
</Root>
```
In fact, comparing D with other node, ratio of D with D compare lower than ratio of D with A compare.
Maybe it is normal behavior, I don't know...

My dirty workaround was mocking Differ.leaf_ratio method like:
```python
def mocked_leaf_ratio(self, left, right):
    ltext = f'<{left.tag}>{self.node_text(left)}</{left.tag}>'
    rtext = f'<{right.tag}>{self.node_text(right)}</{right.tag}>'
    self._sequencematcher.set_seqs(ltext, rtext)
    return self._sequence_ratio()
```
But I neither like this workaround nor the fact of using mock.

**Comment by regebro** [comment]
Well, there is at least two things here that's not ideal, I'll take a look at it.

**Comment by regebro** [comment]
This seems to long ago have been fixed, as I can't reproduce this even with v2.4.
